### PR TITLE
🎉(project) initialize the project

### PR DESCRIPTION
**Pull request by madmatah**
Add python project requirements, django project structure and QA tools.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  # Build backend development environment
+  build-back:
+    docker:
+      - image: circleci/python:3.8-buster
+    working_directory: ~/fun
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-back-dependencies-{{ .Revision }}
```
Comment (jmaupetit) [inline .circleci/config.yml]: If you don't need to invalidate some cache, you can reset the counter.

```suggestion
            - v1-back-dependencies-{{ .Revision }}
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+      - run:
+          name: Install development dependencies
+          command: pip install --user .[dev,sandbox]
+      - save_cache:
+          paths:
+            - ~/.local
+          key: v2-back-dependencies-{{ .Revision }}
```
Comment (jmaupetit) [inline .circleci/config.yml]:
```suggestion
          key: v1-back-dependencies-{{ .Revision }}
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+  lint-back:
+    docker:
+      - image: circleci/python:3.8-buster
+    working_directory: ~/fun
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-back-dependencies-{{ .Revision }}
```
Comment (jmaupetit) [inline .circleci/config.yml]:
```suggestion
            - v1-back-dependencies-{{ .Revision }}
```

File: sandbox/settings.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+    # Database
+    DATABASES = {
+        "default": {
+            "ENGINE": values.Value(
+                "django.db.backends.postgresql_psycopg2",
+                environ_name="DB_ENGINE",
+                environ_prefix=None,
+            ),
+            "NAME": values.Value("richie", environ_name="DB_NAME", environ_prefix=None),
```
Comment (jmaupetit) [inline sandbox/settings.py]:
```suggestion
            "NAME": values.Value("ashley", environ_name="DB_NAME", environ_prefix=None),
```

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+# Switch back to the root user to install development dependencies
+USER root:root
+
+WORKDIR /app
+
+# Copy all sources, not only runtime-required files
+COPY . /app/
+
+# Uninstall richie and re-install it in editable mode along with development
```
Comment (jmaupetit) [inline Dockerfile]:
```suggestion
# Uninstall ashley and re-install it in editable mode along with development
```

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+CMD cd sandbox && \
+    python manage.py runserver 0.0.0.0:8000
+
+# ---- Production image ----
+FROM core as production
+
+WORKDIR /app/sandbox
+
+# The default command runs gunicorn WSGI server in the sandbox
+CMD gunicorn -c /usr/local/etc/gunicorn/richie.py wsgi:application
```
Comment (jmaupetit) [inline Dockerfile]:
```suggestion
CMD gunicorn -c /usr/local/etc/gunicorn/ashley.py wsgi:application
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+.PHONY: lint-flake8
+
+lint-isort: ## automatically re-arrange python imports in back-end code base
+	@echo 'lint:isort started…'
+	@$(COMPOSE_TEST_RUN_APP) isort --recursive --atomic .
+.PHONY: lint-isort
+
+lint-pylint: ## lint back-end python sources with pylint
+	@echo 'lint:pylint started…'
+	@$(COMPOSE_TEST_RUN_APP) pylint src/ashley tests
```
Comment (jmaupetit) [inline Makefile]:
```suggestion
	@$(COMPOSE_TEST_RUN_APP) pylint src/ashley sandbox tests
```

**Comment by madmatah** [comment]
Thanks for your feedback and the wonderful gif @jmaupetit  !

File: sandbox/settings.py [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+"""
+
+import os
+
+from configurations import Configuration, values
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# pylint: disable = W0232
```
Comment (jmaupetit) [inline sandbox/settings.py]: It would be wonderful to add a comment explaining why you choose to silent this pylint warning. :pray:

Comment (madmatah) [reply sandbox/settings.py]: Done :slightly_smiling_face:

File: setup.cfg [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    __pycache__,
+    node_modules,
+    */migrations/*
+
+[isort]
+known_ralph=ralph
+include_trailing_comma=True
+line_length=88
+multi_line_output=3
+sections=FUTURE,STDLIB,THIRDPARTY,RALPH,FIRSTPARTY,LOCALFOLDER
```
Comment (jmaupetit) [inline setup.cfg]:
```suggestion
sections=FUTURE,STDLIB,THIRDPARTY,ASHLEY,FIRSTPARTY,LOCALFOLDER
```

File: setup.cfg [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    .git,
+    .venv,
+    build,
+    venv,
+    __pycache__,
+    node_modules,
+    */migrations/*
+
+[isort]
+known_ralph=ralph
```
Comment (jmaupetit) [inline setup.cfg]:
```suggestion
known_ralph=ashley
```

File: setup.cfg [hunk 11]
```diff
@@ -0,0 +1,86 @@
+[metadata]
+name = ashley
+version = 0.0.1
+description = A self-hosted message board for OpenEdx
```
Comment (jmaupetit) [inline setup.cfg]: I would suggest adding the "forum" keyword.

```suggestion
description = A self-hosted alternative discussion forum for OpenEdx
```
The same applies for the project's README and github project description

**Pull request merged**